packages/components/src/text/next/text-utils.js

```diff
@@ -116,6 +123,7 @@ export function createHighlighterText( {
 					? Object.assign( {}, highlightStyle, activeStyle )
 					: highlightStyle;
 
+			/** @type {Record<string, any>} */
```

Comment:
Is this better than the inferred type? It's a pretty weak type and uses `any`, I'm curious why it needed to be added.

Reply:
Becuase of `highlightIndex` being conditionally added below... I could update it to read:
```js
* @type {{
	children: string;
	className: string;
	key: number;
	style: import('react').CSSProperties;
	"data-g2-component": any;
	highlightIndex?: number;
}} */
const props = {
	...ui.$( 'TextHighlight' ),
	children: text,
	className: highlightClassNames,
	key: index,
	style: highlightStyles,
};
```

But that felt really silly for something that is just being passed directly to `createElement` two statements below.